### PR TITLE
changed deprecated checkbox in favor of Checkbox

### PR DESCRIPTION
**Pull request by akp111**
## Explanation
This PR is raised as part of issue https://github.com/MetaMask/metamask-extension/issues/20164. I have made changes in two files as of now.
`ui/components/app/connected-accounts-permissions/connected-accounts-permissions.js`
and 
`ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-defaults/advanced-gas-fee-defaults.js`

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-defaults/advanced-gas-fee-defaults.js [hunk 1]
```diff
@@ -16,7 +15,7 @@ import { setAdvancedGasFee } from '../../../../store/actions';
 import { useGasFeeContext } from '../../../../contexts/gasFee';
 import { useAdvancedGasFeePopoverContext } from '../context';
 import { useI18nContext } from '../../../../hooks/useI18nContext';
-import { Text } from '../../../component-library';
+import { Text, Checkbox as CheckBox } from '../../../component-library';
```
Comment (georgewrmarshall) [inline ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-defaults/advanced-gas-fee-defaults.js:18]: I think we can remove the alias and update the jsx component?
```suggestion
import { Text, Checkbox } from '../../../component-library';
```

File: ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-defaults/advanced-gas-fee-defaults.js [hunk 2]
```diff
@@ -81,10 +80,10 @@ const AdvancedGasFeeDefaults = () => {
     >
       <label className="advanced-gas-fee-defaults__label">
```
Comment (georgewrmarshall) [inline ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-defaults/advanced-gas-fee-defaults.js:81]: We may be able to reduce some of the jsx here and use the `Checkbox` component API for the label https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-checkbox--docs#label

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey @akp111, thanks for your contribution I would recommend breaking this PR up into smaller PRs for easier review. Would it be possible to submit a PR with just the changes in `ui/checkbox` in one and the changes in `ui/components/app/connected-accounts-permissions/connected-accounts-permissions.js` and `ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-defaults/advanced-gas-fee-defaults.js` in another. You could even break those up into separate PRs as I believe there will need to be some e2e test updates as well so there could be some added complexity to the PR

**Comment by akp111** [comment]
> Hey @akp111, thanks for your contribution I would recommend breaking this PR up into smaller PRs for easier review. Would it be possible to submit a PR with just the changes in `ui/checkbox` in one and the changes in `ui/components/app/connected-accounts-permissions/connected-accounts-permissions.js` and `ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-defaults/advanced-gas-fee-defaults.js` in another. You could even break those up into separate PRs as I believe there will need to be some e2e test updates as well so there could be some added complexity to the PR

Sure! let me do that right away

File: ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-defaults/advanced-gas-fee-defaults.js (hunk 1 above)
Comment (akp111) [reply ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-defaults/advanced-gas-fee-defaults.js:18]: Got it. I did that just so as to follow the naming convection that already exists

**Pull request closed without merging**